druid/src/core.rs

```diff
@@ -272,23 +279,21 @@ impl<T, W: Widget<T>> WidgetPod<T, W> {
     /// [`Rect`]: struct.Rect.html
     /// [`Size`]: struct.Size.html
     /// [`LifeCycle::Size`]: enum.LifeCycle.html#variant.Size
-    pub fn set_origin(&mut self, ctx: &mut LayoutCtx, data: &T, env: &Env, origin: Point) {
-        self.state.origin = origin;
+    pub fn set_origin<'a>(
+        &mut self,
+        ctx: &mut impl ChangeCtx<'a>,
+        _data: &T,
+        _env: &Env,
+        origin: Point,
+    ) {
+        //TODO: decide whether we should keep data and env for compatibility or do a breaking change
```

Comment:
I say go for the breaking change. This is a significant change under the hood and we don't need to keep pretending it didn't happen. Especially as this change is really easy to deal with - you just delete some parameters from your call sites.

What's more, I think we might be able to also delete the `ctx` parameter, see my other comment about `merge_up()` in this method.

Reply:
> I say go for the breaking change. This is a significant change under the hood and we don't need to keep pretending it didn't happen. Especially as this change is really easy to deal with - you just delete some parameters from your call sites.

Agreed :)